src/bg/user-script-registry.js

```diff
@@ -92,6 +99,11 @@ async function loadUserScripts() {
     };
   }).then(loadDetails => {
     let savePromises = loadDetails.map(details => {
+      if (details.iconBlob && ! (details.iconBlob instanceof Blob)) {
+        let info = details.iconBlob;
+        details.iconBlob = new Blob([info.buffer], {'type': info.type});
```

Comment:
Why do we need to do this both here and line 68-72?  Does one or the other only handle updating things already saved the other way?

Reply:
@arantius the 68-72 portion deals with the instance where we search the database for a script with an existing id in order to update it. Before we can update the just fetched script we need to change the array buffer into a blob. 

The other is for when loading the scripts from initial start. Before we do anything with them we transform the array buffer into a blob. 

Both instance ensure we don't do anything if the saved object contains a blob (meaning we're still in the progress of migrating). 

The lines at 68-72 wouldn't be required if the script wasn't fetched from the database. But instead found from the already loaded scripts. 

Either way, once the icons are stored as array buffers you need to convert them to blobs as soon as they're fetched from the database so you don't have inconsistencies or weird errors in other places.